src/python/pants/backend/python/tasks2/python_create_distributions.py

```diff
@@ -29,53 +32,64 @@ def product_types(cls):
   @classmethod
   def prepare(cls, options, round_manager):
     round_manager.require_data(PythonInterpreter)
-    round_manager.require_data('python')  # For codegen.
 
   @staticmethod
   def is_distribution(target):
     return isinstance(target, PythonDistribution)
 
-  def __init__(self, *args, **kwargs):
-    super(PythonCreateDistributions, self).__init__(*args, **kwargs)
+  @property
+  def local_dists_workdir(self):
+    return os.path.join(self.workdir, 'local_dists')
+
+  @property
+  def cache_target_dirs(self):
+    return True
 
   def execute(self):
     dist_targets = self.context.targets(self.is_distribution)
     built_dists = set()
     
     if dist_targets:
-      # Check for duplicate distribution names, since we write the pexes to <dist>/<name>.pex.
-      names = {}
-      for dist_target in dist_targets:
-        name = dist_target.name
-        if name in names:
-          raise TaskError('Cannot build two dist_targets with the same name in a single invocation. '
-                          '{} and {} both have the name {}.'.format(dist_target, names[name], name))
-        names[name] = dist_target
-
-      with self.invalidated(dist_targets, invalidate_dependents=True) as invalidation_check:
+      local_dists_workdir = self.local_dists_workdir
```

Comment:
no need for local_dists_workdir any more.

Reply:
✅